exporter/sematextexporter/es.go

```diff
@@ -0,0 +1,144 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package sematextexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/sematextexporter"
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+	"time"
+
+	json "github.com/json-iterator/go"
+	"github.com/olivere/elastic"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+)
+
+// artificialDocType designates a syntenic doc type for ES documents
+const artificialDocType = "_doc"
+
+type group struct {
+	client *elastic.Client
+	token  string
+}
+
+type client struct {
+	clients map[string]group
+	config  *Config
+	logger  *logrus.Logger
+	writer  FlatWriter
+}
+
+// Client represents a minimal interface client implementation has to satisfy.
+type Client interface {
+	Bulk(body any, config *Config) error
+}
+
+// NewClient creates a new instance of ES client that internally stores a reference
+// to both, event and log receivers.
+func newClient(config *Config, logger *logrus.Logger, writer FlatWriter) (Client, error) {
+	clients := make(map[string]group)
+
+	// client for shipping to logsene
+	if config.LogsConfig.AppToken != "" {
+		c, err := elastic.NewClient(elastic.SetURL(config.LogsEndpoint), elastic.SetSniff(false), elastic.SetHealthcheckTimeout(time.Second*2))
+		if err != nil {
+			return nil, err
+		}
+		clients[config.LogsEndpoint] = group{
+			client: c,
+			token:  config.LogsConfig.AppToken,
+		}
+	}
+
+	return &client{
+		clients: clients,
+		config:  config,
+		logger:  logger,
+		writer:  writer,
+	}, nil
+}
+
+// Bulk processes a batch of documents and sends them to the specified LogsEndpoint.
+func (c *client) Bulk(body any, config *Config) error {
+	if grp, ok := c.clients[config.LogsEndpoint]; ok {
+		bulkRequest := grp.client.Bulk()
+		if reflect.TypeOf(body).Kind() == reflect.Slice {
+			v := reflect.ValueOf(body)
+			for i := 0; i < v.Len(); i++ {
+				doc := v.Index(i).Interface()
+				if docMap, ok := doc.(map[string]any); ok {
+					docMap["os.host"] = getHostname()
```

Comment:
Instead of calling the getHostname func everytime lets save this in the client struct itself as a field similar to what the flinkClient does